module/VuFind/src/VuFind/View/Helper/Root/JsIcons.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * JsIcons helper for passing icon HTML to Javascript
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+/**
+ * JsIcons helper for passing icon HTML to Javascript
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class JsIcons extends JsTranslations
+{
+    /**
+     * Icon helper
+     *
+     * @var Icon
+     */
+    protected $iconHelper;
+
+    /**
+     * Constructor
+     *
+     * @param Icon   $iconHelper Icon helper
+     * @param string $varName    Variable name to store icons
+     */
+    public function __construct(
+        Icon $iconHelper,
+        $varName = 'vufindIconString'
+    ) {
+        $this->iconHelper = $iconHelper;
+        $this->varName = $varName;
```

Comment:
This seems to be unused; can it be removed?

Reply:
Yes. It was included for completeness with JsTranslations but it isn't used by JsIcons.

Reply:
Oh, sorry, I didn't catch that JsIcons was a subclass of JsTranslations. If there are elements of JsTranslations that don't apply to JsIcons, I wonder if we'd be better off refactoring so that the two classes have a common base class instead. (I can do some deeper analysis if you'd find that helpful... I haven't really taken a close look at what this is doing yet).